lib/game.rb

```diff
@@ -1,59 +1,66 @@
-    @cpu_cruiser = Ship.new("Cruiser", 3)
-    @cpu_sub = Ship.new("Submarine", 2)
+    generate_ships_hash
   end
 
+  def generate_ships_hash
+    @ships = Hash.new
+
+    players = { :player => '', :cpu => '' }
+
+    players.each do |key, value|
+      @ships[key] = {
+        submarine: Ship.new("Submarine", 2),
+        cruiser: Ship.new("Cruiser", 3)
+      }
+    end
+    @ships
+  end
 
   # Randomly chooses which algorithm to generate random coordinates
   # and place ships on computer's board
   def cpu_board_setup
-    generator = rand(2)
+    cpu_cruiser = @ships[:cpu][:cruiser]
+    cpu_sub = @ships[:cpu][:submarine]
     start_time = Time.now
+    branch_decision = rand(2)
 
-    if generator == 0
+    if branch_decision == 0
       algorithm = "Merali's"
-      cruiser_coordinates = merali_algorithm(@cpu_board, @cpu_cruiser)
-      @cpu_board.place(@cpu_cruiser, cruiser_coordinates)
-      sub_coordinates = merali_algorithm(@cpu_board, @cpu_sub)
-      @cpu_board.place(@cpu_sub, sub_coordinates)
+      @cpu_board.place(cpu_cruiser, merali_algorithm(@cpu_board, cpu_cruiser))
+      @cpu_board.place(cpu_sub, merali_algorithm(@cpu_board, cpu_sub))
     elsif generator == 1
       algorithm = "Griffith's"
-      cruiser_coordinates = griffith_algorithm(@cpu_board, create_coordinate_array(@cpu_board, @cpu_cruiser))
-      @cpu_board.place(@cpu_cruiser, cruiser_coordinates)
-      sub_coordinates = griffith_algorithm(@cpu_board, create_coordinate_array(@cpu_board, @cpu_sub))
-      @cpu_board.place(@cpu_sub, sub_coordinates)
+      @cpu_board.place(cpu_cruiser, griffith_algorithm(@cpu_board, cpu_cruiser))
+      @cpu_board.place(cpu_sub, griffith_algorithm(@cpu_board, cpu_sub))
     end
-
-    sort_time = Time.now - start_time
-    puts "It took me #{sort_time} seconds to place my two ships according to #{algorithm} algorithm.\n"
+    puts "It took me #{Time.now - start_time} seconds to place my two ships according to #{algorithm} algorithm.\n"
   end
 
 
   def player_board_setup
+    cruiser = @ships[:player][:cruiser]
+    sub = @ships[:player_ships][:sub]
+
     puts "You now need to lay out your two ships."
     puts "The Cruiser is three units long and the Submarine is two units long."
 
     @player_board.render
     puts "Enter the squares for the Cruiser (3 spaces): "
-    @player_board.place(@player_cruiser, get_user_coordinates(@player_cruiser))
-    @player_board.render(true)
+    @player_board.place(cruiser, get_player_coordinates(cruiser))
 
+    @player_board.render(true)
     puts "Enter the squares for the Submarine (2 spaces): "
-    @player_board.place(@player_sub, get_user_coordinates(@player_sub))
+    @player_board.place(sub, get_player_coordinates(sub))
   end
 
 
-  # Validates user's coordinates for ship placement
-  def get_user_coordinates(ship)
+  # Validates player's coordinates for ship placement
+  def get_player_coordinates(ship)
     correct = false
 
     while correct == false
```

Comment:
Do we need to do a nested while + if statement here? it looks like we're only iterating over valid response in this one (not checking for already-shot-shot) so I think aa single while loop could do it, right? 

```ruby
response = gets.chomp
while @player_board.valid_placement?(ship, response) == false
    puts "Those are invalid coordinates. Please try again"
    response = gets.chomp
end
```

Reply:
Or are we just missing the piece here that checks if the shot has been fired already? I think this is where we had put it in but I don't see the check for it?

Reply:
Good refactor! Implemented!